tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php

```diff
@@ -195,4 +199,33 @@ private function assertLimitResult(
             self::assertCount(count($expectedResults), $data);
         }
     }
+
+    public function testLimitWhenOrderByWithSubqueryWithOrderBy(): void
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        if ($platform instanceof DB2Platform) {
+            self::markTestSkipped('Generated SQL is invalid');
```

Comment:
Is it a bug or DB2 just doesn't support LIMIT in subqueries? If it's a bug, the test should be marked as incomplete, not skipped.

Reply:
I don't know this engine.
If I set a limit on the query, I get a DBALException:
<details>
  <summary>Error</summary>

  ```
1) Doctrine\Tests\DBAL\Functional\ModifyLimitQueryTest::testLimitWhenOrderByWithSubqueryWithOrderBy
Exception: [Doctrine\DBAL\Exception] An exception occurred while executing 'SELECT db22.* FROM (SELECT db21.*, ROW_NUMBER() OVER() AS DC_ROWNUM FROM (SELECT test_int FROM modify_limit_table2 ORDER BY (SELECT db22.* FROM (SELECT db21.*, ROW_NUMBER() OVER() AS DC_ROWNUM FROM (SELECT test_int FROM modify_limit_table2 ORDER BY test_int ASC) db21) db22 WHERE db22.DC_ROWNUM <= 1) ASC) db21) db22 WHERE db22.DC_ROWNUM <= 10':

db2_execute(): Statement Execute Failed

With queries:
5. SQL: 'SELECT db22.* FROM (SELECT db21.*, ROW_NUMBER() OVER() AS DC_ROWNUM FROM (SELECT test_int FROM modify_limit_table2 ORDER BY (SELECT db22.* FROM (SELECT db21.*, ROW_NUMBER() OVER() AS DC_ROWNUM FROM (SELECT test_int FROM modify_limit_table2 ORDER BY test_int ASC) db21) db22 WHERE db22.DC_ROWNUM <= 1) ASC) db21) db22 WHERE db22.DC_ROWNUM <= 10' Params: 
4. SQL: 'INSERT INTO modify_limit_table2 (test_int) VALUES (?)' Params: '2'
3. SQL: 'INSERT INTO modify_limit_table2 (test_int) VALUES (?)' Params: '1'
2. SQL: 'TRUNCATE modify_limit_table2 IMMEDIATE' Params: 
1. SQL: 'TRUNCATE modify_limit_table IMMEDIATE' Params: 

Trace:
/home/runner/work/dbal/dbal/lib/Doctrine/DBAL/DBALException.php:159
/home/runner/work/dbal/dbal/lib/Doctrine/DBAL/Connection.php:2121
/home/runner/work/dbal/dbal/lib/Doctrine/DBAL/Connection.php:1264
/home/runner/work/dbal/dbal/lib/Doctrine/DBAL/Connection.php:935
/home/runner/work/dbal/dbal/tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php:185
/home/runner/work/dbal/dbal/tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php:213
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:1536
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:1142
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestResult.php:730
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:883
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestSuite.php:669
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestSuite.php:669
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestSuite.php:669
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:673
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/TextUI/Command.php:148
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/TextUI/Command.php:101
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/phpunit:61


/home/runner/work/dbal/dbal/tests/Doctrine/Tests/DbalFunctionalTestCase.php:110
  ```
</details>
However, when I do not limit the result, I have a warning in `db2_fetch_assoc()`:
<details>
  <summary>Error</summary>

  ```
  1) Doctrine\Tests\DBAL\Functional\ModifyLimitQueryTest::testLimitWhenOrderByWithSubqueryWithOrderBy
Exception: [PHPUnit\Framework\Error\Warning] db2_fetch_assoc(): Fetch Failure

With queries:
5. SQL: 'SELECT db22.* FROM (SELECT db21.*, ROW_NUMBER() OVER() AS DC_ROWNUM FROM (SELECT test_int FROM modify_limit_table2 ORDER BY (SELECT test_int FROM modify_limit_table2 ORDER BY test_int ASC) ASC) db21) db22 WHERE db22.DC_ROWNUM <= 10' Params: 
4. SQL: 'INSERT INTO modify_limit_table2 (test_int) VALUES (?)' Params: '2'
3. SQL: 'INSERT INTO modify_limit_table2 (test_int) VALUES (?)' Params: '1'
2. SQL: 'TRUNCATE modify_limit_table2 IMMEDIATE' Params: 
1. SQL: 'TRUNCATE modify_limit_table IMMEDIATE' Params: 

Trace:
/home/runner/work/dbal/dbal/lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php:300
/home/runner/work/dbal/dbal/lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php:356
/home/runner/work/dbal/dbal/lib/Doctrine/DBAL/Connection.php:935
/home/runner/work/dbal/dbal/tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php:185
/home/runner/work/dbal/dbal/tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php:215
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:1536
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:1142
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestResult.php:730
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:883
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestSuite.php:669
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestSuite.php:669
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/Framework/TestSuite.php:669
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:673
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/TextUI/Command.php:148
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/src/TextUI/Command.php:101
/home/runner/work/dbal/dbal/vendor/phpunit/phpunit/phpunit:61
  ```
</details>